File: 2015WorksApplications/internship/intern/Worksap/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/room/RoomType.java
package jp.co.worksap.intern.entities.room;

public enum RoomType {
	SINGLE, SEMIDOUBLE, DOUBLE;

	public String toString() {
		switch (this) {
		case SINGLE:
			return "SINGLE";
		case SEMIDOUBLE:
			return "SEMIDOUBLE";
		case DOUBLE:
			return "DOUBLE";
		default:
			return ""; 
		}
	}

	public static RoomType valueOfString(String src) {
		String raw = src.toLowerCase();
		if (raw.equals("double")) {
			return DOUBLE;
		}

		if (raw.equals("semidouble")) {
			return SEMIDOUBLE;
		}

		if (raw.equals("single")) {
			return SINGLE;
		}

		throw new IllegalArgumentException("Unknown Room Type : " + raw);
	}
}

File: 2015WorksApplications/internship/b032/Final_submission/VS2012Solution/src/entities/room/room_type.cpp
#include "room_type.h"
#include <string>
#include <stdexcept>
using namespace std;

string room_type_to_string(RoomType type) {
	switch (type) {
	case SINGLE:
		return "SINGLE";
	case SEMIDOUBLE:
		return "SEMIDOUBLE";
	case DOUBLE:
		return "DOUBLE";
	default:
		return "";
	}
}

RoomType room_type_value_of_string(string src) {
	// convert the string, src, in its upper case
	for (string::iterator it = src.begin(); it != src.end(); it++) {
		*it = toupper(*it);
	}

	if (src == "DOUBLE") {
		return DOUBLE;
	}

	if (src == "SEMIDOUBLE") {
		return SEMIDOUBLE;
	}

	if (src == "SINGLE") {
		return SINGLE;
	}

	throw invalid_argument("Unknown Room Type : " + src);
}

